devito/finite_differences/differentiable.py

```diff
@@ -347,6 +348,19 @@ def _eval_is_zero(self):
         return None
 
 
+class DifferentiableFunction(DifferentiableOp):
+
+    def __new__(cls, *args, **kwargs):
+        return cls.__sympy_class__.__new__(cls, *args, **kwargs)
```

Comment:
so here we're 100% sure we have devito.Mul/Add/... otherwise we would get an AttributeError on `__sympy_class__`

Reply:
Elementary function don't have a `__base__` not need diffify (I'm pretty sure) and Mul/Add never ends up here.